src/stdlib/parse_influxdb.rs

```diff
@@ -0,0 +1,332 @@
+use std::collections::BTreeMap;
+
+use chrono::DateTime;
+use influxdb_line_protocol::{FieldValue, ParsedLine};
+
+use crate::compiler::prelude::*;
+
+fn influxdb_lines_to_metrics(line: ParsedLine) -> Vec<ObjectMap> {
+    let ParsedLine {
+        series,
+        field_set,
+        timestamp,
+    } = line;
+
+    let timestamp = timestamp.map(DateTime::from_timestamp_nanos);
+
+    let tags: Option<ObjectMap> = series.tag_set.as_ref().map(|tags| {
+        tags.iter()
+            .map(|t| (t.0.to_string().into(), t.1.to_string().into()))
+            .collect()
+    });
+
+    field_set
+        .iter()
+        .filter_map(|f| {
+            let mut metric = ObjectMap::new();
+            let measurement = &series.measurement;
+            let field_key = f.0.to_string();
+            let field_value = match f.1 {
+                FieldValue::I64(v) => v as f64,
+                FieldValue::U64(v) => v as f64,
+                FieldValue::F64(v) => v,
+                FieldValue::Boolean(v) => {
+                    if v {
+                        1.0
+                    } else {
+                        0.0
+                    }
+                }
+                // TODO: return none or return Error?
+                FieldValue::String(_) => return None,
+            };
+
+            // influxdb_line_protocol crate seems to not allow NaN float values while parsing
+            // field values and this case should not happen, but just in case, we should
+            // ignore the field.
+            let Ok(field_value) = NotNan::new(field_value) else {
+                // TODO: return none or return error?
+                return None;
```

Comment:
Same as before, should we silently ignore or return an error?

I don't think this case is even possible, as I couldn't parse a `NaN`  field value with `influxdb_line_protocol` crate.

Reply:
Mmm, good question. Looking at the parser I agree that there doesn't seem to be a case where it would return NaN. However, this could change in the future. I lean towards returning an error rather than silently ignoring.